### PR TITLE
Cassandra backend

### PR DESCRIPTION
**Pull request by jaron-tielemans**
Keyspace
`CREATE KEYSPACE example WITH replication = {'class': 'SimpleStrategy', 'replication_factor': '1'}  AND durable_writes = true;`

Table
`CREATE TABLE test (key varchar PRIMARY KEY, value varchar);`

Example config where Backend:1 is Cassandra

```
{
    "Backend": "cassandra",
    "Cassandra": {
        "ProtoVersion": 3,
        "Keyspace": "example",
        "Table": "test",
        "Hosts": ["127.0.0.1"],
        "Consistency": 4
    }
}
```

### CODE REVIEW

File: persist-service/cassandra_backend.go [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+}
+
+type CassandraBackend struct {
+	session *gocql.Session
+	config CassandraConf
+}
+
+func (b CassandraBackend) Put(k []byte, v []byte) (bool, error) {
+	err := b.session.Query(fmt.Sprintf(`INSERT INTO %s (key, value) VALUES (?, ?)`, b.config.Table), k, v).Exec()
+	return true, err;
```
Comment (RobinUS2) [inline persist-service/cassandra_backend.go]: Instead of return `true` we should return `err == nil`

File: persist-service/cassandra_backend.go [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+func (b CassandraBackend) Put(k []byte, v []byte) (bool, error) {
+	err := b.session.Query(fmt.Sprintf(`INSERT INTO %s (key, value) VALUES (?, ?)`, b.config.Table), k, v).Exec()
+	return err == nil, err;
+}
+
+func (b CassandraBackend) Get(k []byte) ([]byte, error) {
+	var v []byte;
+	err := b.session.Query(fmt.Sprintf(`SELECT value FROM %s WHERE key = ? LIMIT 1`, b.config.Table), k).Scan(&v)
+	return v, err
```
Comment (RobinUS2) [inline persist-service/cassandra_backend.go]: Same as above

File: persist-service/conf.go [hunk 3]
```diff
@@ -8,12 +8,15 @@ import (
 )
 
 type Conf struct {
+	Backend BackendType
```
Comment (RobinUS2) [inline persist-service/conf.go:11]: Backend 1 is not really user friendly or understandable and depends on the order of the const which is tricky. I'd rather have "Backend": "cassandra" or "file", probably need to add some logic into the newConf()

Comment (jaron-tielemans) [reply persist-service/conf.go:11]: BackendType is now a string, see updated config and iBackend.go

File: persist-service/cassandra_backend.go [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+}
+
+func newCassandraBackend(conf *Conf) IBackend  {
+	// Connect to the cluster
+	cluster := gocql.NewCluster(conf.Cassandra.Hosts...)
+	cluster.Keyspace = conf.Cassandra.Keyspace
+	cluster.Consistency = conf.Cassandra.Consistency
+	cluster.ProtoVersion = conf.Cassandra.ProtoVersion
+	session, _ := cluster.CreateSession()
```
Comment (RobinUS2) [inline persist-service/cassandra_backend.go]: We should handle this error

**Comment by RobinUS2** [comment]
@FlxJaron can you also add something in the readme? Basically what you put in the description of this PR is a good start.

**Comment by RobinUS2** [comment]
Looks good 👍

**Pull request merged**